bundles/org.openhab.binding.openweathermap/src/main/resources/OH-INF/thing/channel-types.xml

```diff
@@ -407,6 +413,32 @@
 		<state readOnly="true" pattern="%1$tY-%1$tm-%1$td %1$tH:%1$tM:%1$tS"/>
 	</channel-type>
 
+	<channel-type id="moonrise">
+		<item-type>DateTime</item-type>
+		<label>Moonrise Time</label>
+		<description>Time of moonrise for the given day.</description>
+		<category>Time</category>
+		<state readOnly="true" pattern="%1$tY-%1$tm-%1$td %1$tH:%1$tM:%1$tS"/>
+	</channel-type>
+
+	<channel-type id="moonset">
+		<item-type>DateTime</item-type>
+		<label>Moonset Time</label>
+		<description>Time of moonset for the given day.</description>
+		<category>Time</category>
+		<state readOnly="true" pattern="%1$tY-%1$tm-%1$td %1$tH:%1$tM:%1$tS"/>
+	</channel-type>
+
+	<channel-type id="moon_phase">
+		<item-type>Number</item-type>
+		<label>Moon Phase</label>
+		<description>Moon phase for the given day. 0 and 1 are 'new moon', 0.25 is 'first quarter moon', 0.5 is 'full moon'
+			and 0.75 is 'last quarter moon'. The periods in between are called 'waxing crescent', 'waxing gibous', 'waning
+			gibous', and 'waning crescent', respectively.</description>
+		<category>Moon</category>
+		<state readOnly="true" pattern="%.2f"/>
```

Comment:
It should be possible to add state description options, so openHAB automatically displays 'new moon' etc., but the periods in between could be an issue.

Reply:
I was wondering about this, but decided to kick the can down the road :)

I do assume that there won't be periods in between the 4 stages. Won't it operate similar to a digital clock, which for example shows 8h43 for 60 seconds, until it's 8h44? (Similarly, I expect the moon-phase (MOON_PHASE? ;)) to be "0.25" for 7 days and then turn to "0.5".

But as to how that would show as "waning crescent" etc. in openHAB... No idea (yet).

Reply:
I understand the API docs in a way that the moon phase float can be between 0.25 and 0.5, the closer you get to half moon, the close that float gets to 0.5.

State description options (https://www.openhab.org/docs/developer/bindings/thing-xml.html#state-description) only work for equality, so I don’t think they are very useful here.

Reply:
You could however add an additional channel that does the “translation”.

Reply:
But I’d first focus on getting this PR finished before considering this.